### PR TITLE
Added Option to log all players after reorder

### LINKED ISSUE #3959: 18CZ log doesn't include full setting of priority
https://18xx.games/game/27956?action=159

[17:38]OFE's share price changes from K50 to K53
[17:38]beardbru has priority deal
[17:38]-- Operating Round 3.1 (of 1) --

The log only shows who was given priority, not who got position 2, 3, etc. The log should probably just say:

[time]New priority order: beardbru, ChrisTheCat, papavk, Corian09, tgg

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Made it a little bit more generic. Maybe other games can use the default implementation more

CHanged the display of prioty buttons for most_cash orderings to the 1828 style

closes #3959

### CODE REVIEW

File: lib/engine/game/base.rb [hunk 1]
```diff
@@ -2076,7 +2076,11 @@ def reorder_players(order = nil)
           current_order = @players.dup
           @players.sort_by! { |p| [p.cash, current_order.index(p)] }
         end
-        @log << "#{@players.first.name} has priority deal"
+        @log << if log_all_players
+                  "Priority order: #{@players.reject(&:bankrupt).map(&:name).join(', ')}"
```
Comment (crericha) [inline lib/engine/game/base.rb:2080]: 1822 has this feature. Does it make sense to use that?

**Comment by crericha** [comment]
FYI 1828.Games added turn order to the player cards. It could be something you want to take advantage of. For me personally, I rather look there than scroll up in the log to find the order.

image

**Comment by Zwergenpunk** [comment]
oh wow thanks. I want that :D

**Comment by crericha** [comment]
> oh wow thanks. I want that :D
> 
> Maybe the log is still good, but your feature sounds really good.

One thing to know is the "Next SR" order shows during the previous SR too as that is relevant for 1828. You'll either want to update the next SR order after each stock action or keep the list empty until the end of the stock round (in which case it won't display).

**Comment by Zwergenpunk** [comment]
> > oh wow thanks. I want that :D
> > Maybe the log is still good, but your feature sounds really good.
> 
> One thing to know is the "Next SR" order shows during the previous SR too as that is relevant for 1828. You'll either want to update the next SR order after each stock action or keep the list empty until the end of the stock round (in which case it won't display until you want it to).

Thanks for the info. 
I think it is ok that its empty during SR. It does not give any information which player has which position at a given time during SR.

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -2061,7 +2061,7 @@ def next_sr_player_order
         self.class::NEXT_SR_PLAYER_ORDER
       end
 
-      def reorder_players(order = nil)
+      def reorder_players(order = nil, log_all_players: false)
```
Comment (crericha) [inline lib/engine/game/base.rb:2064]: A bit of a nitpick, but would 'log_player_order' make more sense for the name of the kwarg?

**Review by crericha — APPROVED** [review]

**Pull request merged**